common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java

```diff
@@ -240,6 +310,12 @@ public Mono<Void> runAllAsync(long endNanoTime) {
      */
     abstract Mono<Integer> runTestAsync();
 
+    abstract CompletableFuture<Integer> runTestAsyncWithCompletableFuture();
+
+    abstract Runnable runTestAsyncWithExecutorService();
+
+    abstract void runTestAsyncWithVirtualThread();
```

Comment:
Since these aren't mainline scenarios yet, may want to have base class implementations where these throw exceptions / return error states.